doc/source/whatsnew/v1.1.0.rst

```diff
@@ -1169,6 +1169,7 @@ Other
 - Bug in :class:`Tick` comparisons raising ``TypeError`` when comparing against timedelta-like objects (:issue:`34088`)
 - Bug in :class:`Tick` multiplication raising ``TypeError`` when multiplying by a float (:issue:`34486`)
 - Passing a `set` as `names` argument to :func:`pandas.read_csv`, :func:`pandas.read_table`, or :func:`pandas.read_fwf` will raise ``ValueError: Names should be an ordered collection.`` (:issue:`34946`)
+- Improved error message for invalid construction (:issue:`35190`)
```

Comment:
Could you mention the object that caused the invalid construction?

Reply:
Improved error message for invalid addition of index? (To be honest that is as clear as mud)

I'll be honest I don't know that I understand what you're asking here.

Reply:
Yeah that would be better. 

I'd just be clearer to know `Improved error message for invalid construction` of what?